src/main/java/com/prgrms/needit/domain/user/email/service/EmailService.java

```diff
@@ -78,45 +86,45 @@ private MimeMessage createMessage(String receiver, String code) {
 	}
 
 	public void sendMessage(String receiver) {
-		if (isEmailExist(receiver)) {
+		if (redisTemplate.opsForValue()
+						 .get("RT:" + receiver) != null) {
 			throw new DuplicatedResourceException(ErrorCode.ALREADY_EXIST_EMAIL);
 		}
 
-		final String key = createKey();
-		MimeMessage message = createMessage(receiver, key);
+		final String code = createCode();
+		redisTemplate.opsForValue()
+					 .set(
+						 "RT:" + receiver, "EC:" + code, EXPIRATION, TimeUnit.MILLISECONDS
+					 );
```

Comment:
앞부분에 RT 를 붙이신 이유가 궁금합니다. 저는 refresh token 인 것을 구분시키기 위하여 붙였거든요!! 
그리고 하정님의 이메일 인증코드 만료에 대해서는 인증코드가 주이기 때문에 저는 인증코드가 key값으로 들어가고 사용자 이메일이 value값으로 들어가야하지 않을까 생각됩니다.

```suggestion
		redisTemplate.opsForValue()
					 .set(
						"EC:" + code, receiver, EXPIRATION, TimeUnit.MILLISECONDS
					 );
```

Reply:
아니면 이메일로 구분시키는 것이 맞으니 레디스에서 **EC**로 refresh token(RT)와 구분을 두어서 key값으로 EC + 사용자 이메일로 넣고 value 값으로 인증코드를 관리하는 게 맞는 것 같아용!!